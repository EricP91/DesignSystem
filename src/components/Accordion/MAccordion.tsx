import React from 'react';
import {
  Accordion,
  AccordionDetails,
  AccordionProps,
  AccordionSummary as MuiAccordionSummary,
  styled,
} from '@mui/material';

export interface MAccordionProps extends AccordionProps {
  header: React.ReactNode;
  expandIcon?: React.ReactNode;
}

const AccordionSummary = styled(MuiAccordionSummary)(() => ({
  '& .MuiAccordionSummary-content': {
    margin: 0,
  },
}));

const MAccordion = ({
  header,
  children,
  expandIcon,
  disableGutters = true,
  ...props
}: MAccordionProps): JSX.Element => (
  <Accordion disableGutters={disableGutters} {...props}>
    <AccordionSummary expandIcon={expandIcon}>{header}</AccordionSummary>
    <AccordionDetails>{children}</AccordionDetails>
  </Accordion>
);

export default MAccordion;
