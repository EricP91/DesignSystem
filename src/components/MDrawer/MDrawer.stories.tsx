import { Story } from '@storybook/react';
import React, { useRef } from 'react';
import { Box } from '@mui/material';
import MDrawer, { MDrawerProps } from './MDrawer';

export default {
  title: 'Components/Drawer',
  parameters: {
    layout: 'fullscreen',
  },
};

const Template: Story<MDrawerProps> = (args) => {
  const ref = useRef();
  return (
    <Box
      ref={ref}
      sx={{
        position: 'relative',
        width: '100%',
        background: 'url(https://img.wallpapersafari.com/desktop/1920/1080/80/77/sEl8VZ.jpg) no-repeat fixed center',
        height: '100%',
      }}
    >
      <MDrawer ref={ref} {...args} />
    </Box>
  );
};

export const Default = Template.bind({});
Default.args = {
  drawerWidth: 280,
  children: <Box sx={{ backgroundColor: '#F1F4F6', height: '100%', textAlign: 'center', p: 2 }}>Drawer Content</Box>,
};
