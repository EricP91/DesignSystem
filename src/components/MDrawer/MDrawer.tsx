import * as React from 'react';
import { forwardRef, MutableRefObject, useEffect } from 'react';
import { Box, Drawer, DrawerProps, useTheme } from '@mui/material';
import { DrawerToggleIcon } from '../../assets/icons';

export interface MDrawerProps extends Omit<DrawerProps, 'css'> {
  drawerWidth: number;
  open?: boolean;
  onOpenChange?: (value: boolean) => void | undefined;
}
const MDrawer = forwardRef(
  ({ drawerWidth, children, open = false, onOpenChange, ...otherProps }: MDrawerProps, ref): JSX.Element => {
    const [isOpen, setIsOpen] = React.useState<boolean>(open);
    const theme = useTheme();

    const enterScreenTransition = theme.transitions.create('transform', {
      duration: theme.transitions.duration.enteringScreen,
      easing: theme.transitions.easing.easeOut,
    });
    const leaveScreenTransition = theme.transitions.create('transform', {
      duration: theme.transitions.duration.leavingScreen,
      easing: theme.transitions.easing.sharp,
    });

    useEffect(() => {
      if (open !== isOpen) setIsOpen(open);
      // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [open]);

    const toggleDrawer = (): void => {
      if (onOpenChange) {
        onOpenChange(!isOpen);
      } else {
        setIsOpen(!isOpen);
      }
    };

    const transformX = isOpen ? `translateX(${drawerWidth}px)` : 'translateX(0)';
    const rotateDegree = isOpen ? 180 : 0;

    return (
      <Box sx={{ display: 'flex', height: '100%', width: '100%' }}>
        <Box
          onClick={toggleDrawer}
          sx={{
            transition: isOpen ? enterScreenTransition : leaveScreenTransition,
            borderTopRightRadius: '4px',
            borderBottomRightRadius: '4px',
            height: 56,
            width: 32,
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
            backgroundColor: 'white',
            position: 'relative',
            top: '50%',
            transform: `translateY(-50%) ${transformX}`,
            cursor: 'pointer',
            zIndex: 1,
          }}
        >
          <DrawerToggleIcon
            data-testid="toggle-icon"
            sx={{ height: 14, width: 8, transform: `rotate(${rotateDegree}deg)` }}
          />
        </Box>
        <Drawer
          variant="persistent"
          anchor="left"
          open={isOpen}
          PaperProps={{ style: { position: 'absolute', width: drawerWidth } }}
          SlideProps={{ container: (ref as MutableRefObject<HTMLElement>)?.current, style: { position: 'absolute' } }}
          {...otherProps}
        >
          {children}
        </Drawer>
      </Box>
    );
  }
);
export default MDrawer;
