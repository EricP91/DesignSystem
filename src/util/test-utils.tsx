import React, { FC, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeConfig } from '../theme';

const AllTheProviders: FC = ({ children }): JSX.Element => <ThemeConfig isLightMode>{children}</ThemeConfig>;

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
