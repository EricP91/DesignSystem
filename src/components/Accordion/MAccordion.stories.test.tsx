import { render } from 'test-utils';
import React from 'react';
import { Default } from '../MDrawer/MDrawer.stories';
import { Single } from './MAccordion.stories';
import { MAccordionProps } from './MAccordion';

it('should render MAccordion story', () => {
  const story = render(<Single {...(Default.args as MAccordionProps)} />);
  expect(story).toMatchSnapshot();
});
