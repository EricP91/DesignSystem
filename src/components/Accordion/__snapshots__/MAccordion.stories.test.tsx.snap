// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MAccordion story 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded css-nr5z1c-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root css-8e8s2s-MuiButtonBase-root-MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
          />
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                class="MuiAccordion-region"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-7wcoc8-MuiAccordionDetails-root"
                >
                  <div
                    class="MuiBox-root css-lvvxb2"
                  >
                    Drawer Content
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded css-nr5z1c-MuiPaper-root-MuiAccordion-root"
    >
      <div
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root css-8e8s2s-MuiButtonBase-root-MuiAccordionSummary-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
        />
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiAccordion-region"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root css-7wcoc8-MuiAccordionDetails-root"
              >
                <div
                  class="MuiBox-root css-lvvxb2"
                >
                  Drawer Content
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
